packages/esm-patient-chart-app/src/visit/visits-widget/visit.resource.tsx

```diff
@@ -5,28 +5,45 @@ import {
   useOpenmrsInfinite,
   type OpenmrsResource,
   type Visit,
-  makeUrl,
   useOpenmrsPagination,
 } from '@openmrs/esm-framework';
 import { type ChartConfig } from '../../config-schema';
 
 const customRepresentation =
   'custom:(uuid,location,encounters:(uuid,diagnoses:(uuid,display,rank,diagnosis,voided),form:(uuid,display),encounterDatetime,orders:full,obs:(uuid,concept:(uuid,display,conceptClass:(uuid,display)),display,groupMembers:(uuid,concept:(uuid,display),value:(uuid,display),display),value,obsDatetime),encounterType:(uuid,display,viewPrivilege,editPrivilege),encounterProviders:(uuid,display,encounterRole:(uuid,display),provider:(uuid,person:(uuid,display)))),visitType:(uuid,name,display),startDatetime,stopDatetime,patient,attributes:(attributeType:ref,display,uuid,value)';
 
-export function useInfiniteVisits(patientUuid: string) {
+export function useInfiniteVisits(
+  patientUuid: string,
+  params: Record<string, number | string> = {},
+  rep: string = customRepresentation,
+) {
   const { numberOfVisitsToLoad } = useConfig<ChartConfig>();
 
-  const url = `${restBaseUrl}/visit?patient=${patientUuid}&v=${customRepresentation}&limit=${numberOfVisitsToLoad}`;
+  const url = new URL(
+    `${window.openmrsBase}/${restBaseUrl}/visit?patient=${patientUuid}&v=${rep}&limit=${numberOfVisitsToLoad}`,
+    window.location.toString(),
+  );
+  for (const key in params) {
+    url.searchParams.set(key, '' + params[key]);
+  }
+
   const { data, ...rest } = useOpenmrsInfinite<Visit>(patientUuid ? url : null);
 
   return { visits: data, ...rest };
 }
 
-export function usePaginatedVisits(patientUuid: string, pageSize: number) {
+export function usePaginatedVisits(
+  patientUuid: string,
+  pageSize: number,
+  params: Record<string, number | string> = {},
+) {
   const url = new URL(
-    makeUrl(`${restBaseUrl}/visit?patient=${patientUuid}&v=${customRepresentation}`),
+    `${window.openmrsBase}/${restBaseUrl}/visit?patient=${patientUuid}&v=${customRepresentation}`,
```

Comment:
It's preferential to use `getOpenmrsSpaBase()`, which just guarantees that there is always a `/` at the end of the `spaBase`. The `spaBase` variable itself is whatever was defined at build time, which may or may not have a trailing slash.

```suggestion
    `${window.getOpenmrsSpaBase()}${restBaseUrl}/visit?patient=${patientUuid}&v=${customRepresentation}`,
```